src/cosma/cinterface.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <mpi.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
It would be great to have a comment here explaining what these variables inside `layout` represent. Maybe we can even add some concrete example. This is a great feature that will certainly be used and it's important we have this clarified.

Reply:
I've added some comments, let's add the official docs somewhere when the interface is fixed